client/src/App.js

```diff
@@ -1,13 +1,28 @@
 import React from 'react';
 import { Provider } from 'react-redux';
-import store from './store';
-import Landing from './components/layout/Landing';
+import { ConnectedRouter } from 'connected-react-router';
+import { Switch, Route } from 'react-router-dom';
+
+import configureStore, { history } from './store';
+import Home from './components/Home';
+import Landing from './components/landing/Landing';
+import Layout from './components/layout/Layout';
+import PrivateRoute from './components/layout/PrivateRoute';
 import './App.css';
 
+const store = configureStore({});
+
 function App() {
   return (
     <Provider store={store}>
-      <Landing />
+      <ConnectedRouter history={history}>
+        <Layout>
+          <Switch>
+            <Route exact path='/' component={Landing} />
+            <PrivateRoute exact path='/home' component={Home} />
+          </Switch>
+        </Layout>
+      </ConnectedRouter>
```

Comment:
Can you fill me in on what's happening here?

Reply:
Okay let's start with the routing. `ConnectedRouter` is the parent component which allows us to navigate to various routes and render the specified component when the route is matched. The `Switch` matches routes exclusively and the react router teams explains it better than I can:
>`<Switch>` is unique in that it renders a route exclusively. In contrast, every `<Route>` that matches the location renders inclusively.
Then we have the two routes, one of which is private. `PrivateRoute` is a custom component here that will redirect to the login page if there is no user (in the future we'd check a token's validity or something). These two routes are displaying the `Landing` and `Home` components respectively. Make sense?